services/workflows-service/src/workflow/workflow.service.ts

```diff
@@ -1776,10 +1777,11 @@ export class WorkflowService {
 
     const { id } = await this.endUserService.create({
       data: {
-        correlationId: entity.id,
+        correlationId: entity.id ? entity.id : undefined,
```

Comment:
```suggestion
        correlationId: entity.id ?? undefined,
```

Reply:
`??` won't work here, I wanted to ensure that if `entity.id` is an empty string, then use `undefined` instead of the empty string. I could make this condition to make it more specific so it will be clear in the future why we added it in the first place.

Reply:
So ‘||’?